functional-programming-spreadsheet/index680.js

```diff
@@ -0,0 +1,102 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const spreadsheetFunctions = {
+  "": x => x
+};
+
+/*
+The array destructuring syntax can be used to extract values from arrays:
+```
+const [x, y] = [1, 2]; // in variables
+const fn = ([x, y]) => x + y // in functions
+```
+Use this syntax to define a function `random` in `spreadsheetFunctions` which takes an array of two elements and returns the first one.
```

Comment:
Use this syntax to define a function `random` in `spreadsheetFunctions` which takes the array `[x, y]` and returns the the first element.